qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -969,16 +961,29 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                 val_mapping={'ON': 1, 'OFF': 0},
                                 vals=vals.Enum('ON', 'OFF') )
 
-            self.add_parameter('signal_output{}_enable'.format(sigout),
-                                label="Enable signal output's amplitude.",
-                                set_cmd=partial(self._sigout_setter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                get_cmd=partial(self._sigout_getter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                val_mapping={'ON': 1, 'OFF': 0},
-                                vals=vals.Enum('ON', 'OFF') )
+            for output in range(1, 9):
+                self.add_parameter(
+                    'signal{}_output{}_enable'.format(sigout, output),
+                    label="Enable signal output's amplitude.",
+                    set_cmd=partial(self._sigout_setter,
+                                    sigout - 1, 0,
+                                    'enables/{}'.format(output - 1)),
+                    get_cmd=partial(self._sigout_getter,
+                                    sigout - 1, 0,
+                                    'enables/{}'.format(output - 1)),
+                    val_mapping={'ON': 1, 'OFF': 0},
```

Comment:
for all the on/off parameters, you might want to use [`create_on_off_val_mapping` function from](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/utils/helpers.py#L670), [read more about it here](http://qcodes.github.io/Qcodes/examples/Creating%20Instrument%20Drivers.html?highlight=drivers#On/Off-parameters).

Reply:
I will have a look.

Reply:
✔️